explore/exploration.h

```diff
@@ -0,0 +1,207 @@
+#pragma once
+
+#include <stdint.h>
+#include <stdexcept>
+#include <algorithm>
+#include <numeric>
+#include <cstring>
+
+#define S_EXPLORATION_OK                0
+#define E_EXPLORATION_BAD_RANGE         1
+
+namespace exploration
+{
+  template<typename It>
+  int generate_epsilon_greedy(float epsilon, uint32_t top_action, It pdf_first, It pdf_last, std::random_access_iterator_tag pdf_tag)
+  {
+    if (pdf_last < pdf_first)
+      return E_EXPLORATION_BAD_RANGE;
+
+    size_t num_actions = pdf_last - pdf_first;
+    if (num_actions == 0)
+      return S_EXPLORATION_OK;
```

Comment:
This doesn't seem ok, because you aren't returning a normalized distribution.

Reply:
done